eshell-info-banner.el

```diff
@@ -314,12 +314,26 @@ If RELEASE-FILE is nil, use '/etc/os-release'."
   "Read the operating system information from lsb_release"
   (shell-command-to-string "lsb_release -d -s"))
 
+(defun eshell-info-banner--get-os-information-from-registry ()
+  "Read the operating system information from the Windows registry."
+  (let ((win32-name "Windows")
+        (win32-build "Unknown"))
+    (with-temp-buffer
+      (call-process "reg" nil t nil "query" "HKLM\\Software\\Microsoft\\Windows NT\\CurrentVersion")
+      (goto-char (point-min))
+      (while (re-search-forward "\\([^[:blank:]]+\\) *\\(REG_[^[:blank:]]+\\) *\\(.+\\)" nil t)
+        (cond
+         ((string= "ProductName" (match-string 1)) (setq win32-name (match-string 3)))
+         ((string= "BuildLab" (match-string 1)) (setq win32-build (match-string 3)))))
```

Comment:
Short note about `setq`. I looked it up in the [manual, `setq` only changes the local] `win32-name` and `win32-build` values and doesn't introduce global/buffer local ones.

 [1]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Setting-Variables.html

Reply:
I find the documentation really not clear about that, but I’ll trust you on that. I’ll modify it if I find out it does actually set it globally.